Compilation failure under clang with std=gnu2x

At least one of the C source files in cfitsio (`iraffits.c`) uses function definitions with identifier lists, something that is seemingly being removed in C23 ([ref1](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2432.pdf), [ref2](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2841.htm)).

Replacing these function definitions shouldn't be too hard, and they don't see to be used broadly across the codebase (compilation failed after successfully compiling 60 files), so I'd be happy to put together a PR.

Originally seen while compiling [Rfits](https://github.com/asgr/Rfits) on a MacOS system with clang-1700.0.13.5, MacOSx15.5 SDK. But also reproduced on an Ubuntu 24.10 machine with clang-19.

Full failure log (generated in the Linux machine):

```
$> ./configure CC="clang-19 -std=gnu2x"
....
$> make all
...
iraffits.c:1310:9: error: unknown type name 'hstring'
 1310 | hgeti4 (hstring,keyword,ival)
      |         ^
iraffits.c:1310:17: error: unknown type name 'keyword'
 1310 | hgeti4 (hstring,keyword,ival)
      |                 ^
iraffits.c:1310:25: error: unknown type name 'ival'
 1310 | hgeti4 (hstring,keyword,ival)
      |                         ^
iraffits.c:1310:30: error: expected ';' after top level declarator
 1310 | hgeti4 (hstring,keyword,ival)
      |                              ^
      |                              ;
iraffits.c:1320:1: error: expected identifier or '('
 1320 | {
      | ^
iraffits.c:1355:8: error: unknown type name 'hstring'
 1355 | hgets (hstring, keyword, lstr, str)
      |        ^
iraffits.c:1355:17: error: unknown type name 'keyword'
 1355 | hgets (hstring, keyword, lstr, str)
      |                 ^
iraffits.c:1355:26: error: unknown type name 'lstr'
 1355 | hgets (hstring, keyword, lstr, str)
      |                          ^
iraffits.c:1355:32: error: unknown type name 'str'
 1355 | hgets (hstring, keyword, lstr, str)
      |                                ^
iraffits.c:1355:36: error: expected ';' after top level declarator
 1355 | hgets (hstring, keyword, lstr, str)
      |                                    ^
      |                                    ;
iraffits.c:1366:1: error: expected identifier or '('
 1366 | {
      | ^
iraffits.c:1394:8: error: unknown type name 'hstring'
 1394 | hgetc (hstring,keyword0)
      |        ^
iraffits.c:1394:16: error: unknown type name 'keyword0'
 1394 | hgetc (hstring,keyword0)
      |                ^
iraffits.c:1394:25: error: expected ';' after top level declarator
 1394 | hgetc (hstring,keyword0)
      |                         ^
      |                         ;
iraffits.c:1403:1: error: expected identifier or '('
 1403 | {
      | ^
iraffits.c:1545:11: error: unknown type name 'hstring'
 1545 | blsearch (hstring,keyword)
      |           ^
iraffits.c:1545:19: error: unknown type name 'keyword'
 1545 | blsearch (hstring,keyword)
      |                   ^
iraffits.c:1545:27: error: expected ';' after top level declarator
 1545 | blsearch (hstring,keyword)
      |                           ^
      |                           ;
iraffits.c:1561:1: error: expected identifier or '('
 1561 | {
      | ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

--- a/iraffits.c
+++ b/iraffits.c
@@ -1307,16 +1307,16 @@ static int lhead0 = 0;
 /* Extract long value for variable from FITS header string */
 
 static int
-hgeti4 (hstring,keyword,ival)
-
-char *hstring;	/* character string containing FITS header information
-		   in the format <keyword>= <value> {/ <comment>} */
-char *keyword;	/* character string containing the name of the keyword
-		   the value of which is returned.  hget searches for a
-		   line beginning with this string.  if "[n]" is present,
-		   the n'th token in the value is returned.
-		   (the first 8 characters must be unique) */
-int *ival;
+hgeti4 (
+    char *hstring, /* character string containing FITS header information
+                      in the format <keyword>= <value> {/ <comment>} */
+    char *keyword, /* character string containing the name of the keyword
+                      the value of which is returned.  hget searches for a
+                      line beginning with this string.  if "[n]" is present,
+                      the n'th token in the value is returned.
+                      (the first 8 characters must be unique) */
+    int *ival
+)
 {
 char *value;
 double dval;
@@ -1352,17 +1352,17 @@ char val[30];
 /* Extract string value for variable from FITS header string */
 
 static int
-hgets (hstring, keyword, lstr, str)
-
-char *hstring;	/* character string containing FITS header information
-		   in the format <keyword>= <value> {/ <comment>} */
-char *keyword;	/* character string containing the name of the keyword
-		   the value of which is returned.  hget searches for a
-		   line beginning with this string.  if "[n]" is present,
-		   the n'th token in the value is returned.
-		   (the first 8 characters must be unique) */
-int lstr;	/* Size of str in characters */
-char *str;	/* String (returned) */
+hgets (
+  char *hstring,  /* character string containing FITS header information
+                     in the format <keyword>= <value> {/ <comment>} */
+  char *keyword,  /* character string containing the name of the keyword
+                     the value of which is returned.  hget searches for a
+                     line beginning with this string.  if "[n]" is present,
+                     the n'th token in the value is returned.
+                     (the first 8 characters must be unique) */
+  int lstr,       /* Size of str in characters */
+  char *str       /* String (returned) */
+)
 {
 	char *value;
 	int lval;
@@ -1391,15 +1391,15 @@ char *str;	/* String (returned) */
 /* Extract character value for variable from FITS header string */
 
 static char *
-hgetc (hstring,keyword0)
-
-char *hstring;	/* character string containing FITS header information
-		   in the format <keyword>= <value> {/ <comment>} */
-char *keyword0;	/* character string containing the name of the keyword
-		   the value of which is returned.  hget searches for a
-		   line beginning with this string.  if "[n]" is present,
-		   the n'th token in the value is returned.
-		   (the first 8 characters must be unique) */
+hgetc (
+  char *hstring, /* character string containing FITS header information
+                    in the format <keyword>= <value> {/ <comment>} */
+  char *keyword0 /* character string containing the name of the keyword
+                    the value of which is returned.  hget searches for a
+                    line beginning with this string.  if "[n]" is present,
+                    the n'th token in the value is returned.
+                    (the first 8 characters must be unique) */
+)
 {
 	static char cval[80];
 	char *value;
@@ -1541,23 +1541,22 @@ char *keyword0;	/* character string containing the name of the keyword
 /*-------------------------------------------------------------------*/
 /* Find beginning of fillable blank line before FITS header keyword line */
 
-static char *
-blsearch (hstring,keyword)
-
 /* Find entry for keyword keyword in FITS header string hstring.
    (the keyword may have a maximum of eight letters)
    NULL is returned if the keyword is not found */
-
-char *hstring;	/* character string containing fits-style header
-		information in the format <keyword>= <value> {/ <comment>}
-		the default is that each entry is 80 characters long;
-		however, lines may be of arbitrary length terminated by
-		nulls, carriage returns or linefeeds, if packed is true.  */
-char *keyword;	/* character string containing the name of the variable
-		to be returned.  ksearch searches for a line beginning
-		with this string.  The string may be a character
-		literal or a character variable terminated by a null
-		or '$'.  it is truncated to 8 characters. */
+static char *
+blsearch (
+  char *hstring, /* character string containing fits-style header
+                    information in the format <keyword>= <value> {/ <comment>}
+                    the default is that each entry is 80 characters long;
+                    however, lines may be of arbitrary length terminated by
+                    nulls, carriage returns or linefeeds, if packed is true.  */
+  char *keyword  /* character string containing the name of the variable
+                    to be returned.  ksearch searches for a line beginning
+                    with this string.  The string may be a character
+                    literal or a character variable terminated by a null
+                    or '$'.  it is truncated to 8 characters. */
+)
 {
     char *loc, *headnext, *headlast, *pval, *lc, *line;
     char *bval;
@@ -1638,22 +1637,21 @@ char *keyword;	/* character string containing the name of the variable
 /*-------------------------------------------------------------------*/
 /* Find FITS header line containing specified keyword */
 
-static char *ksearch (hstring,keyword)
-
 /* Find entry for keyword keyword in FITS header string hstring.
    (the keyword may have a maximum of eight letters)
    NULL is returned if the keyword is not found */
-
-char *hstring;	/* character string containing fits-style header
-		information in the format <keyword>= <value> {/ <comment>}
-		the default is that each entry is 80 characters long;
-		however, lines may be of arbitrary length terminated by
-		nulls, carriage returns or linefeeds, if packed is true.  */
-char *keyword;	/* character string containing the name of the variable
-		to be returned.  ksearch searches for a line beginning
-		with this string.  The string may be a character
-		literal or a character variable terminated by a null
-		or '$'.  it is truncated to 8 characters. */
+static char *ksearch (
+  char *hstring, /* character string containing fits-style header
+                    information in the format <keyword>= <value> {/ <comment>}
+                    the default is that each entry is 80 characters long;
+                    however, lines may be of arbitrary length terminated by
+                    nulls, carriage returns or linefeeds, if packed is true.  */
+  char *keyword  /* character string containing the name of the variable
+                    to be returned.  ksearch searches for a line beginning
+                    with this string.  The string may be a character
+                    literal or a character variable terminated by a null
+                    or '$'.  it is truncated to 8 characters. */
+)
 {
     char *loc, *headnext, *headlast, *pval, *lc, *line;
     int icol, nextchar, lkey, nleft, lhstr;
@@ -1717,11 +1715,10 @@ char *keyword;	/* character string containing the name of the variable
 /* Find string s2 within null-terminated string s1 */
 
 static char *
-strsrch (s1, s2)
-
-char *s1;	/* String to search */
-char *s2;	/* String to look for */
-
+strsrch (
+  char *s1, /* String to search */
+  char *s2  /* String to look for */
+)
 {
     int ls1;
     ls1 = strlen (s1);
@@ -1732,12 +1729,11 @@ char *s2;	/* String to look for */
 /* Find string s2 within string s1 */
 
 static char *
-strnsrch (s1, s2, ls1)
-
-char	*s1;	/* String to search */
-char	*s2;	/* String to look for */
-int	ls1;	/* Length of string being searched */
-
+strnsrch (
+  char *s1, /* String to search */
+  char *s2, /* String to look for */
+  int ls1   /* Length of string being searched */
+)
 {
     char *s,*s1e;
     char cfirst,clast;
@@ -1797,18 +1793,17 @@ int	ls1;	/* Length of string being searched */
 /*  HPUTI4 - Set int keyword = ival in FITS header string */
 
 static void
-hputi4 (hstring,keyword,ival)
-
-  char *hstring;	/* character string containing FITS-style header
-			   information in the format
-			   <keyword>= <value> {/ <comment>}
-			   each entry is padded with spaces to 80 characters */
-
-  char *keyword;		/* character string containing the name of the variable
-			   to be returned.  hput searches for a line beginning
-			   with this string, and if there isn't one, creates one.
-		   	   The first 8 characters of keyword must be unique. */
-  int ival;		/* int number */
+hputi4 (
+  char *hstring, /* character string containing FITS-style header
+                    information in the format
+                    <keyword>= <value> {/ <comment>}
+                    each entry is padded with spaces to 80 characters */
+  char *keyword, /* character string containing the name of the variable
+                    to be returned.  hput searches for a line beginning
+                    with this string, and if there isn't one, creates one.
+                    The first 8 characters of keyword must be unique. */
+  int ival       /* int number */
+)
 {
     char value[30];
 
@@ -1827,11 +1822,11 @@ hputi4 (hstring,keyword,ival)
 /*  HPUTL - Set keyword = F if lval=0, else T, in FITS header string */
 
 static void
-hputl (hstring, keyword,lval)
-
-char *hstring;		/* FITS header */
-char *keyword;		/* Keyword name */
-int lval;		/* logical variable (0=false, else true) */
+hputl (
+  char *hstring, /* FITS header */
+  char *keyword, /* Keyword name */
+  int lval       /* logical variable (0=false, else true) */
+)
 {
     char value[8];
 
@@ -1853,12 +1848,12 @@ int lval;		/* logical variable (0=false, else true) */
 /*  HPUTS - Set character string keyword = 'cval' in FITS header string */
 
 static void
-hputs (hstring,keyword,cval)
-
-char *hstring;	/* FITS header */
-char *keyword;	/* Keyword name */
-char *cval;	/* character string containing the value for variable
-		   keyword.  trailing and leading blanks are removed.  */
+hputs (
+  char *hstring, /* FITS header */
+  char *keyword, /* Keyword name */
+  char *cval    /* character string containing the value for variable
+                   keyword.  trailing and leading blanks are removed.  */
+)
 {
     char squot = 39;
     char value[70];
@@ -1887,12 +1882,12 @@ char *cval;	/* character string containing the value for variable
 /*  HPUTC - Set character string keyword = value in FITS header string */
 
 static void
-hputc (hstring,keyword,value)
-
-char *hstring;
-char *keyword;
-char *value;	/* character string containing the value for variable
-		   keyword.  trailing and leading blanks are removed.  */
+hputc (
+  char *hstring,
+  char *keyword,
+  char *value    /* character string containing the value for variable
+                    keyword.  trailing and leading blanks are removed.  */
+)
 {
     char squot = 39;
     char line[100];
@@ -2029,11 +2024,7 @@ char *value;	/* character string containing the value for variable
 /*  HPUTCOM - Set comment for keyword or on line in FITS header string */
 
 static void
-hputcom (hstring,keyword,comment)
-
-  char *hstring;
-  char *keyword;
-  char *comment;
+hputcom (char *hstring, char *keyword, char *comment)
 {
 	char squot;
 	char line[100];

